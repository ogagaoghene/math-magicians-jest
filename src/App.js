import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Navbar from './Components/Navbar';
import Home from './Components/Home';
import Calculator from './Components/Calculator';
import Quotes from './Components/Quotes';
import './styles.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="Calculator" element={<Calculator />} />
          <Route path="Quotes" element={<Quotes />} />
        </Routes>
      </>

    );
  }
}

export default App;
