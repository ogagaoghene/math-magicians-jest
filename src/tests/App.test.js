import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom'
import App from '../App'

 it("Check if App component is rendered", () => {
    const tree = renderer.create(
    <Router><App /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
})