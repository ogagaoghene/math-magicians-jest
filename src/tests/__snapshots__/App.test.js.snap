// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if App component is rendered 1`] = `
Array [
  <nav
    className="navigation"
  >
    <div
      className="brand-name"
    >
      Math Magicians
    </div>
    <div>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className=""
            href="/Calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            className=""
            href="/Quotes"
            onClick={[Function]}
          >
            Quotes
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="grid__wrapper"
  >
    <h1>
      Welcome to Our Page!
    </h1>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Amet esse aperiam reprehenderit repudiandae quod vitae? Eius perspiciatis expedita porro quos accusamus odio non, consequatur provident enim, assumenda distinctio aperiam dolores.
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Amet esse aperiam reprehenderit repudiandae quod vitae? Eius perspiciatis expedita porro quos accusamus odio non, consequatur provident enim, assumenda distinctio aperiam dolores.
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Amet esse aperiam reprehenderit repudiandae quod vitae? Eius perspiciatis expedita porro quos accusamus odio non, consequatur provident enim, assumenda distinctio aperiam dolores.
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Amet esse aperiam reprehenderit repudiandae quod vitae? Eius perspiciatis expedita porro quos accusamus odio non, consequatur provident enim, assumenda distinctio aperiam dolores.
    </p>
  </div>,
]
`;
